# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1550183260 -3600
#      Thu Feb 14 23:27:40 2019 +0100
# Branch stable
# Node ID 8e1e5a44a632072ac4a5468b89b105cdd750dd92
# Parent  c66f509d9f5ca52665c2d52157f81ba710b9c6be
colorbar.m: fix regression when peer axes is reset (bug #55713)

* colobar.m (cb_clim): Don't run listener if the colorbar has already been
  deleted.

diff --git a/scripts/plot/draw/colorbar.m b/scripts/plot/draw/colorbar.m
--- a/scripts/plot/draw/colorbar.m
+++ b/scripts/plot/draw/colorbar.m
@@ -438,17 +438,17 @@ function cb_restore_axes (hcb, ~, hax, o
     set (hax, "__colorbar_handle__", []);
   endif
 
 endfunction
 
 ## Update colorbar when clim has changed
 function cb_clim (hax, ~, hcb, hi)
 
-  if (isaxes (hax))
+  if (isaxes (hax) && isaxes (hcb))
     clen = rows (get (hax, "colormap"));
     cext = get (hax, "clim");
     cdiff = (cext(2) - cext(1)) / clen / 2;
     cmin = cext(1) + cdiff;
     cmax = cext(2) - cdiff;
 
     if (strcmp (get (hcb, "__vertical__"), "on"))
       set (hi, "ydata", [cmin, cmax]);
